packages/vscode-dependencies-validation/test/vscodeMocks.ts

```diff
@@ -0,0 +1,189 @@
+    throw new Error("Function not implemented.");
+  },
+  getText: function (range?: Range): string {
+    throw new Error("Function not implemented.");
+  },
+  getWordRangeAtPosition: function (
+    position: Position,
+    regex?: RegExp
+  ): Range | undefined {
+    throw new Error("Function not implemented.");
+  },
+  validateRange: function (range: Range): Range {
+    throw new Error("Function not implemented.");
+  },
+  validatePosition: function (position: Position): Position {
+    throw new Error("Function not implemented.");
+  },
+  // @ts-expect-error -- https://stackoverflow.com/questions/68799234/typescript-pick-only-specific-method-from-overload-to-be-passed-to-parameterst
+  lineAt: function (line: Position): TextLine {
+    throw new Error("Function not implemented.");
+  },
+};
+
+export const position: Position = {
+  line: 0,
+  character: 0,
+  isBefore: function (other: Position): boolean {
+    throw new Error("Function not implemented.");
+  },
+  isBeforeOrEqual: function (other: Position): boolean {
+    throw new Error("Function not implemented.");
+  },
+  isAfter: function (other: Position): boolean {
+    throw new Error("Function not implemented.");
+  },
+  isAfterOrEqual: function (other: Position): boolean {
+    throw new Error("Function not implemented.");
+  },
+  isEqual: function (other: Position): boolean {
+    throw new Error("Function not implemented.");
+  },
+  compareTo: function (other: Position): number {
+    throw new Error("Function not implemented.");
+  },
+  // @ts-expect-error -- https://stackoverflow.com/questions/68799234/typescript-pick-only-specific-method-from-overload-to-be-passed-to-parameterst
+  translate: function (lineDelta?: number, characterDelta?: number): Position {
+    throw new Error("Function not implemented.");
+  },
+  // @ts-expect-error -- https://stackoverflow.com/questions/68799234/typescript-pick-only-specific-method-from-overload-to-be-passed-to-parameterst
+  with: function (line?: number, character?: number): Position {
+    throw new Error("Function not implemented.");
+  },
+};
+
+export const range: Range = {
+  start: position,
+  end: position,
+  isEmpty: false,
+  isSingleLine: false,
+  contains: function (positionOrRange: Position | Range): boolean {
+    throw new Error("Function not implemented.");
+  },
+  isEqual: function (other: Range): boolean {
+    throw new Error("Function not implemented.");
+  },
+  intersection: function (range: Range): Range | undefined {
+    throw new Error("Function not implemented.");
+  },
+  union: function (other: Range): Range {
+    throw new Error("Function not implemented.");
+  },
+  // @ts-expect-error -- https://stackoverflow.com/questions/68799234/typescript-pick-only-specific-method-from-overload-to-be-passed-to-parameterst
+  with: function (start?: Position, end?: Position): Range {
+    throw new Error("Function not implemented.");
+  },
+};
+
+export const context: CodeActionContext = {
```

Comment:
as opposed to VSCode like classes, this looks like static data for a specific test
should it be part of the test ?

Reply:
even at cost of slight duplication?

Reply:
if not, maybe at the very least freeeze it so one test won't modify it for the next one